### ISSUE TITLE
[discussion] One-liner simple processing 

### ISSUE THREAD

**Issue by culebron** [geopandas/geopandas#754]
Hi everyone,
I saw the discussion mailing list is rather inactive, so let me ask a discussional question here:

**TL;DR:** would you like to have a utility to do processing shortcuts for subsetting, slicing, and attribute filtering?

    python3 my_script.py input_file.gpkg:layer_name{column1,column2:rename2,column3}[:100] output_file.gpkg
    python3 my_script.py one-layer-file.gpkg[intersects other-file.gpkg] output_file.gpkg{-useless_column}
    

(Not sure if it should be proposed to GPD.)

## Longer story

It's about a year I use GeoPandas as a primary geodata processing tool instead of PostGIS.  I see there are many chores that I want to fold into minimalistic code, to stop writing python code.

I did write myself a wrapper, to pass geo-files from command line directly into functions (using [Argh](http://argh.readthedocs.io/en/latest/) library). But still it's not enough. 

## Good Side

The scripts that I write look like this:

    from my_lib import autoargs
    import geopandas

    @autoargs
    def main(input_gdf: gpd.GeoDataFrame, param:int=0):
        # do something
        return output_gdf

You can import this function and call it with geodataframe and get it back, and also you can call it as a CLI script. *Argh* will print command line help if args are incorrect. It can also fold parameter to one-letters and bools to flags.

    $ my_buffer_script.py points.gpkg 1000 buffers.gpkg

## Not Enough

When I have a huge dataset and want to do a processing pipeline, I end up either writing python code for filtering, or doing it manually, or writing a Makefile.

    houses-sample.gpkg: houses.gpkg
         my_lib sample $^ 1000 $@
    result.gpkg: houses-sample.gpkg
         python3 do-something.py $^ some params $@

That's still a chore to write those lines, and with one-liner it could be wrapped a bit:

    result.gpkg: houses.gpkg
              python3 do-something.py $^[:1000] some params $@

or 
   
     python3 do-something.py houses.gpkg[intersects sample-district.gpkg] some params result.gpkg
     python3 do-something.py houses.gpkg[city='Novosibirsk'] some params result.gpkg

or, if I have a dict of operations and use them in the notation, it can become like this:

    result.gpkg: houses.gpkg
              python3 do-something.py $^ sample(1000) some params $@

As you can see, square brackets stand for either id, or slice, or attribute filter. Subset can be written with curly braces, including renaming operation:

     my_file.gpkg{column1:rename1,column2,column3}

or possibly delete columns:
 
    my_file.gpkg{-that_useless_column}

What do you think of this?

**Comment by jorisvandenbossche** [comment]
@culebron Thanks for opening the issue. 
Personally, I think we want to keep the scope of geopandas limited to the "python library", and not extend it to a command line interface as well (we are already limited in resources, and designing a full new cli API is not a small thing, plus there are probably many ways to do it as well).

But of course, that doesn't hold back to develop an external CLI package that builds upon geopandas (we would be certainly happy to reference such a package in our documentation). 
Also, if you have specific feedback on the *python* API to make things easier or less verbose, that's certainly welcome as well!

**Comment by martinfleis** [comment]
I will close this issue as out of scope for GeoPandas.
